modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/ddl/DdlSqlToCommandConverter.java

```diff
@@ -322,31 +321,22 @@ private CreateTableCommand convertCreateTable(IgniteSqlCreateTable createTblNode
                     + "querySql=\"" + ctx.query() + "\"]");
         }
 
-        // TODO: https://issues.apache.org/jira/browse/IGNITE-21673 Remove dedupSetPk after this issue is fixed.
-        Set<String> dedupSetPk;
         List<String> pkColumns;
```

Comment:
I think it is better to use this variable directly (assign pkColumns = new ArrayList<>(columnNodes.size());)
since it is not used in any conditional logic.

Reply:
fixed